src/mame/drivers/hp9k_3xx.cpp

```diff
@@ -257,18 +255,16 @@ uint32_t hp9k3xx_state::screen_update(screen_device &screen, bitmap_rgb32 &bitma
 static INPUT_PORTS_START( hp9k330 )
 INPUT_PORTS_END
 
-WRITE_LINE_MEMBER(hp9k3xx_state::cpu_reset)
-{
-}
-
 void hp9k3xx_state::driver_start()
 {
 	m_diag_led.resolve();
 }
 
 void hp9k3xx_state::machine_reset()
 {
-	m_maincpu->set_reset_callback(write_line_delegate(FUNC(hp9k3xx_state::cpu_reset), this));
+	auto *dio = subdevice<bus::hp_dio::dio16_device>("diobus");
+	if (dio)
+		m_maincpu->set_reset_callback(write_line_delegate(FUNC(bus::hp_dio::dio16_device::reset_in), dio));
```

Comment:
It's a really bad idea to be doing this on reset.  Devices should be resolving callback on start, so by this time it should be too late if the CPU is resolving callbacks correctly.  You'll definitely know whether you have a DIO bus or not by configuration complete time so there's no need to wait until reset.  It's also not going to change between resets in a session.  But really, this should be done in machine configuration when the DIO bus is added.